services/src/main/java/org/fao/geonet/api/registries/vocabularies/KeywordsApi.java

```diff
@@ -327,10 +327,7 @@ public Object searchKeywords(
     )
     @RequestMapping(
         path = "/keyword",
-        method = {
-            RequestMethod.GET,
-            RequestMethod.POST
-        },
+        method = RequestMethod.GET,
```

Comment:
The issue is that we are using it for facet translations which can fail using GET when lots of keywords are requested https://github.com/geonetwork/core-geonetwork/commit/d38ce1e83505cc37817ed3358bf4e06fe6359d7e 
So we should probably keep the 2 methods

Reply:
Do you think we should consider removing the GET instead?

It seems odd to create an api and indicate - use Get for small list of parameters and Post for large list of parameters?

If not then I can duplicate the api specification so that we have both.  And I will add the reason.

Reply:
> Do you think we should consider removing the GET instead?

We probably use the GET in the editor.  It is probably better to duplicate it even if it is odd. I don't see better option.

Reply:
@fxprunayre - Would it be ok with keeping both but mark the GET method as depreciated?  This would give the option to remove the GET method in the future and it would also inform people that the POST is preferred.

Reply:
> Would it be ok with keeping both but mark the GET method as depreciated? This would give the option to remove the GET method in the future and it would also inform people that the POST is preferred.

We use it when using xlinks eg. https://github.com/geonetwork/core-geonetwork/blob/main/schemas/iso19139/src/main/plugin/iso19139/convert/thesaurus-transformation.xsl#L151 so here we can only use GET.

Reply:
OK - I added both api's to fix the duplicate operation id.
I called the GET method version `getKeywordById` and the POST method version `getKeywordByIds`.  Open to any suggestions on names.